lib/saluki-components/src/relays/otlp/config.rs

```diff
@@ -0,0 +1,83 @@
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_error::GenericError;
+use saluki_io::net::ListenAddress;
+use serde::Deserialize;
+
+const fn default_grpc_max_recv_msg_size_bytes() -> usize {
+    4 * 1024 * 1024
+}
+fn default_http_endpoint() -> ListenAddress {
+    ListenAddress::Tcp("0.0.0.0:4318".parse().expect("valid socket address"))
+}
+
+fn default_grpc_endpoint() -> ListenAddress {
+    ListenAddress::Tcp("0.0.0.0:4317".parse().expect("valid socket address"))
+}
+
+fn default_otlp_destination_endpoint() -> String {
+    "http://localhost:4319".to_string()
+}
+
+/// Configuration for the OTLP relay.
+#[derive(Deserialize)]
+pub struct OtlpRelayConfiguration {
+    /// The HTTP endpoint to listen on for OTLP requests.
+    ///
+    /// Defaults to `tcp://0.0.0.0:4318`.
+    #[serde(default = "default_http_endpoint")]
+    http_endpoint: ListenAddress,
+
+    /// The gRPC endpoint to listen on for OTLP requests.
+    ///
+    /// Defaults to `tcp://0.0.0.0:4317`.
+    #[serde(default = "default_grpc_endpoint")]
+    grpc_endpoint: ListenAddress,
+
+    /// Maximum size, in bytes, of a gRPC message that can be received.
+    ///
+    /// Defaults to 4 MiB.
+    #[serde(default = "default_grpc_max_recv_msg_size_bytes")]
+    grpc_max_recv_msg_size_bytes: usize,
+
+    /// The destination endpoint to forward OTLP data to.
+    ///
+    /// This is the base URL (e.g., `http://localhost:4319`) where OTLP data will be proxied.
+    /// The relay will append the appropriate paths (`/v1/metrics`, `/v1/logs`).
+    ///
+    /// Defaults to `http://localhost:4319`.
+    #[serde(default = "default_otlp_destination_endpoint")]
+    otlp_destination_endpoint: String,
```

Comment:
Note, I believe we'll need to be able to configure this by signal type since we will want to route traces to trace-agent and the rest to the Core Agent.